Index: src/main/java/it/polimi/ingsw/galaxytruckerproject/cards/MeteorSwarm.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package it.polimi.ingsw.galaxytruckerproject.cards;\r\n\r\nimport it.polimi.ingsw.galaxytruckerproject.FlightBoard;\r\nimport it.polimi.ingsw.galaxytruckerproject.cards.projectiles.Meteor;\r\n\r\nimport java.util.ArrayList;\r\n\r\npublic class MeteorSwarm extends Card {\r\n    private final ArrayList<Meteor> listOfMeteors;\r\n\r\n    public MeteorSwarm(int level, ArrayList<Meteor> listOfMeteors) {\r\n        super(level, 0);\r\n        this.listOfMeteors = listOfMeteors;\r\n    }\r\n\r\n    @Override\r\n    public void executeCard(FlightBoard flightBoard) {\r\n        listOfMeteors.forEach((meteor) -> meteor.throwMeteor(flightBoard.getRanking()));\r\n    }\r\n}\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/it/polimi/ingsw/galaxytruckerproject/cards/MeteorSwarm.java b/src/main/java/it/polimi/ingsw/galaxytruckerproject/cards/MeteorSwarm.java
--- a/src/main/java/it/polimi/ingsw/galaxytruckerproject/cards/MeteorSwarm.java	(revision eb0d7ca9cc280afe212ac311af4ef64c71b66510)
+++ b/src/main/java/it/polimi/ingsw/galaxytruckerproject/cards/MeteorSwarm.java	(date 1741787902024)
@@ -1,5 +1,6 @@
 package it.polimi.ingsw.galaxytruckerproject.cards;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
 import it.polimi.ingsw.galaxytruckerproject.FlightBoard;
 import it.polimi.ingsw.galaxytruckerproject.cards.projectiles.Meteor;
 
@@ -8,6 +9,7 @@
 public class MeteorSwarm extends Card {
     private final ArrayList<Meteor> listOfMeteors;
 
+    @JsonCreator
     public MeteorSwarm(int level, ArrayList<Meteor> listOfMeteors) {
         super(level, 0);
         this.listOfMeteors = listOfMeteors;
Index: src/main/java/it/polimi/ingsw/galaxytruckerproject/Game.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/it/polimi/ingsw/galaxytruckerproject/Game.java b/src/main/java/it/polimi/ingsw/galaxytruckerproject/Game.java
new file mode 100644
--- /dev/null	(date 1741808423390)
+++ b/src/main/java/it/polimi/ingsw/galaxytruckerproject/Game.java	(date 1741808423390)
@@ -0,0 +1,4 @@
+package it.polimi.ingsw.galaxytruckerproject;
+
+public class Game {
+}
